### PR TITLE
Addition of velocity scale factor estimation to EKF

### PR DESCRIPTION
**Pull request by rsasaki0109**
#### What does this implement?
I have added velocity scale factor estimation to `extended_kalman_filter.py`.
This is for correcting the vehicle speed measured with scale factor errors due to factors such as wheel wear.

ekf_animation

#### Additional information
I have adjusted the noise for velocity scale factor estimation, as it was not well-estimated with the default parameters.
It is assumed that the GNSS is RTK FIXed.

#### CheckList
- [ ] Did you add an unittest for your new example or defect fix?
- [x] Did you add documents for your new example?
- [x] All CIs are green? (You can check it after submitting)

### CODE REVIEW

**Comment by AtsushiSakai** [comment]
Hi, thank you for great PR!! 
Is it possible to add a new script for it, not modify the current file?

**Comment by rsasaki0109** [comment]
Thank you for your suggestion.
I've now added the new script `ekf_with_velocity_correction.py` to the `Localization/extended_kalman_filter` directory, which introduces the velocity correction feature for EKF.
Does this look good to you?

**Comment by AtsushiSakai** [comment]
Thank you. Is it possible to update [this doc](https://github.com/AtsushiSakai/PythonRobotics/blob/master/docs/modules/localization/extended_kalman_filter_localization_files/extended_kalman_filter_localization_main.rst) to explain your code too?

**Comment by rsasaki0109** [comment]
OK, I will do that.

**Comment by rsasaki0109** [comment]
@AtsushiSakai 
I have added the document. Is this the correct way to add it?

**Comment by AtsushiSakai** [comment]
I'm sorry for my late reply. @rsasaki0109 
For document, I think it is better to modify this file https://github.com/AtsushiSakai/PythonRobotics/blob/master/docs/modules/localization/extended_kalman_filter_localization_files/extended_kalman_filter_localization_main.rst
instead of adding new docs.

You can do grouping the "Filter design", "Motion model", "Observation model" for only position estimation and your position and velocity estimation.
https://github.com/AtsushiSakai/PythonRobotics/blob/master/docs/modules/localization/extended_kalman_filter_localization_files/extended_kalman_filter_localization_main.rst

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
Only remaining comment is for doc structure.

**Comment by AtsushiSakai** [comment]
@rsasaki0109 Hi. Is it possible to finish this PR? Only remaining task is just organizing docs. If it is difficult for you, I can do it. Let me know. Thanks for good PR.

**Comment by rsasaki0109** [comment]
Sorry, I didn't notice your comment. I'll work on it when I have some free time.

**Comment by rsasaki0109** [comment]
@AtsushiSakai 
Is your comment meant to mean this?

- Motion Model
  - only position estimation
  - position and velocity estimation
 
- Observation model
  - only position estimation
  - position and velocity estimation

**Comment by AtsushiSakai** [comment]
> Is your comment meant to mean this?
Yes. I think other two “filter design” and “Extended Kalman Filter” are common.

**Comment by AtsushiSakai** [comment]
sorry, is it possible to merge the latest master code to build doc in CI?

File: docs/modules/localization/extended_kalman_filter_localization_files/extended_kalman_filter_localization_main.rst [hunk 1]
```diff
@@ -22,9 +26,26 @@ The red ellipse is estimated covariance ellipse with EKF.
...
+
+
+.. image:: ekf_with_velocity_correction_1_0.png
+   :width: 600px
+
+This is a Extended kalman filter (EKF) localization with velocity correction.
+
+This is for correcting the vehicle speed measured with scale factor errors due to factors such as wheel wear.
+
+Code: `PythonRobotics/extended_kalman_ekf_with_velocity_correctionfilter.py at master ﾂｷ
```
Comment (AtsushiSakai) [inline docs/modules/localization/extended_kalman_filter_localization_files/extended_kalman_filter_localization_main.rst:40]:
```suggestion
Code: `PythonRobotics/extended_kalman_ekf_with_velocity_correctionfilter.py
```

**Comment by rsasaki0109** [comment]
Thank you for your assistance! (I'm sorry, I've been a bit busy this month and my response was delayed...)

**Pull request merged**